### PR TITLE
Do not crash when Bluemix datasource are not found

### PR DESCRIPTION
**Pull request by hacksparrow**
#### Related issues

#279 

- None

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by hacksparrow** [comment]
@bajtos @raymondfeng guys, I need your input.

The model generator should not proceed ahead if no Bluemix datasource is defined. I can think of two ways of doing this.

- `return done('No Bluemix datasource found')` - async operation, cannot be asserted conventionally
- `process.exit(1)` - will exit `mocha` too; can't use this, if we want to add test cases

**Comment by hacksparrow** [comment]
@bajtos thanks for the suggestion on Slack, I have made the changes accordingly to include the test case.

File: model/index.js [hunk 1]
```diff
@@ -64,13 +60,41 @@ module.exports = yeoman.Base.extend({
 
   addNullDataSourceItem: actions.addNullDataSourceItem,
 
+  setBluemixDatasourceState: function() {
+    if (this.options.bluemix) {
+      var configPath = path.resolve(process.cwd(), '.bluemix',
```
Comment (raymondfeng) [inline model/index.js]: Let's use `this.destinationPath('.bluemix/datasources-config.json')` instead. See http://yeoman.io/authoring/file-system.html.

File: model/index.js [hunk 2]
```diff
@@ -64,13 +60,41 @@ module.exports = yeoman.Base.extend({
...
+                      'datasources-config.json');
+      if (!fs.existsSync(configPath)) {
+        console.log('datasources-config.json not found');
```
Comment (raymondfeng) [inline model/index.js]: Use `this.log` instead.

File: model/index.js [hunk 3]
```diff
@@ -64,13 +60,36 @@ module.exports = yeoman.Base.extend({
 
   addNullDataSourceItem: actions.addNullDataSourceItem,
 
+  setBluemixDatasourceState: function() {
+    if (this.options.bluemix) {
+      var configPath = this.destinationPath('.bluemix/datasources-config.json');
+      if (!fs.existsSync(configPath)) {
+        this.log('datasources-config.json not found');
+        process.exit();
```
Comment (raymondfeng) [inline model/index.js]: It's not ideal, but I cannot find a graceful way in Yeoman to report the error and terminate the generator. Unfortunately `done(err)` prints ugly stack trace. I'm not sure if `process.exit()` will mess up the terminal.

Comment (hacksparrow) [inline model/index.js]: Yes, I agree with the points.

No prompts are presented when `process.exit()` is called at that point, so it should be a clear exit. What other ways could this mess up the terminal?

Showing `done(err)`'s stack trace for that particular condition doesn't look like good UI. However, I decided to use `done(err)` on line no 85 to make the code testable.

**Pull request merged**